File: cppCode/Three indices.cpp
#include <iostream>
using namespace std;
const int N=1000;

void solve(){
        int n;
        int p[N];
        cin>>n;
        for(int i=0;i<n;i++){
            cin>>p[i];
        }

        for(int i=1;i<n-1;i++){
            if(p[i]>p[i-1] && p[i]>p[i+1]){
   
                cout<<"YES"<<endl;
                cout<<i<<" "<<i+1<<" "<<i+2<<endl;
                return;
            }
        }

        cout<<"NO"<<endl;

}
int main(){
    int tests;
    cin>>tests;
    while(tests){
        solve();
        tests--;
    }
}

File: CodeForces python/Three indices.py
tests=int(input())

for i in range(tests):
    n=int(input())
    p=[int(x) for x in input().split()]
    found="NO"
    for i in range(1,n-1):
        if p[i]>p[i-1] and p[i]>p[i+1]:
            found="YES"
            print("YES")
            print(i,i+1,i+2)
            break
    if found=="NO":
        print("NO")